R/length-params.R

```diff
@@ -0,0 +1,158 @@
+#' Download metadata catch types
+#'
+#' This function download airtable tables containing metadata of catch types.
+#'
+#' @param log_threshold The (standard Apache logj4) log level used as a
+#'   threshold for the logging infrastructure. See [logger::log_levels] for more
+#'   details
+#'
+#' @return
+#' @export
+#'
+get_catch_types <- function(log_threshold = logger::DEBUG){
+
+  pars <- read_config()
+
+  metadata <- get_preprocessed_metadata(pars)
+
+  catch_types <- metadata$catch_types
+
+  fao_records <- air_get_records(table="fao_catch",
+                                 base_id=pars$metadata$airtable$base_id,
+                                 api_key=readLines("auth/airtable-key")) %>%
+    air_records_to_tibble()
+
+  catches_tab <- dplyr::left_join(fao_records,catch_types,by="interagency_code")
+
+  spp <- catches_tab %>%
+    dplyr::select(interagency_code,isscaap_taxocode.x,name_scientific,
+                  catch_family,catch_name_en,taxo_name_en)
+
+  logger::log_info("Getting taxa rank...")
+  ranks <- taxize::tax_rank(spp$name_scientific,db = "gbif",rows=1)
+
+
+  ranks_tab <-
+    ranks %>%
+    purrr::map(~stringr::str_to_title(.x)) %>%
+    purrr::map(~tidyr::as_tibble(.x)) %>%
+    purrr::map_dfr(~dplyr::mutate_all(.x,as.character)) %>%
+    dplyr::rename(taxa_rank=value) %>%
+    dplyr::bind_cols(spp,.) %>%
+    dplyr::mutate(name=name_scientific) %>%
+    dplyr::mutate(name=dplyr::case_when(name=="Thunnini"~"Tuna",
+                                        name=="Rajiformes"~"Manta",
```

Comment:
I don't think Mantas are Rajiformes. Actually this makes me think that we might have chosen the wrong taxocode here as Rajiformes are skates but I would believe that stingrays are also often captured in Timor. Do you have any insight @altilley?

Reply:
Yes @efcaguab that's a point I didn't highlight, happy that you took it. There is some confusion on the fao_catch table, do unnamed record rows in the fao_catch table should be removed?